src/daemon/control.rs

```diff
@@ -454,6 +455,37 @@ fn share_unvault_signatures(
     send_sig_msg(&mut transport, txid, sigs.clone())
 }
 
+// Derive the DepositTxIn (to be referenced in an Unvault transaction) out of an existing
+// deposit outpoint.
+fn deposit_txin_from_outpoint(
+    revaultd: &RevaultD,
+    outpoint: &OutPoint,
+) -> Result<Result<(DepositTxIn, ChildNumber), RpcControlError>, ControlError> {
+    match db_vault_by_deposit(&revaultd.db_file(), outpoint)? {
```

Comment:
I was thinkgin more like:
```rust
fn deposit_txin_from_vault(
    revaultd: &RevaultD,
    vault: &Vault,
) -> Result<DepositTxIn, RpcControlError> {
        match vault.status {
            VaultStatus::Active => {
                let deposit_descriptor = revaultd.deposit_descriptor.derive(vault.derivation_index);
                Ok((
                    DepositTxIn::new(
                        *outpoint,
                        DepositTxOut::new(
                            vault.amount.as_sat(),
                            &deposit_descriptor,
                            revaultd.xpub_ctx(),
                        ),
                    ))
            }
            status => Err(RpcControlError::InvalidStatus((
                status,
                VaultStatus::Active,
            ))),
    }
}
```

and call it here:
``` rust
          for outpoint in outpoints.iter() {
                    let vault = match db_vault_by_deposit(&revaultd.db_file(), outpoint)? {
                       Ok(vault) => vault,
                       None => response_tx.send(RpcControlError::UnknownOutpoint(*outpoint))?
                    }
                    let deposit_txin= deposit_txin_from_vault(&revaultd, &vault)
                        match deposit_txin_from_outpoint(&revaultd, &outpoint)? {
                            Ok(res) => res,
                            Err(e) => {
                                response_tx.send(Err(e))?;
                                continue;
                            }
                        };
                    if vault.derivation_index > change_index {
                        change_index = vault.derivation_index;
                    }
                    txins.push((deposit_txin, vault.derivation_index));
                }
```

Reply:
Ok, it's actually super misguided. Just going to revert the previous change and go with that.

Reply:
ok for me